### PR TITLE
Det's starting revolver and ammo are now MOSTLY non-lethal (and adds new .38 rubber bullets)

### PR DESCRIPTION
**Pull request by TeomanTheGreat**
## About The Pull Request
Det now starts with rubber bullets instead of their lethal ones.

## Why It's Good For The Game
Detectives (as per the sop) shouldn't be performing arrests and them having lethal rounds just encourages them to blast people (which they already do). Mostly non-lethal rounds should fix this problem to an extent and it takes above 2 full clips plus what's already starting in the revolver to kill someone with these new rounds.

## Changelog
:cl: Phil Smith
balance: Detective's now start with rubber/bouncy rubber revolver bullets in their gun and shoulder holster
/:cl:

### CODE REVIEW

**Comment by Archanial** [comment]
I love how we ported AI datums but then didn't port any features associated with it nor fixes

**Comment by FriendlyContractor** [comment]
Making detectives weapon non-lethal will not discourage detective from performing arrests at all. Also dying doesnt matter in this game you can always be quickly fixed up with no consequences

**Comment by mcmeiler** [comment]
Seems like a duplicate. We can already make rubber shots at autolathes. Just that detective players ~~have no idea how to use an autolathe~~ forget to print them and instead use lethals.
- #4504

**Comment by TeomanTheGreat** [comment]
> Seems like a duplicate. We can already make rubber shots at autolathes. Just that detective players ~~have no idea how to use an autolathe~~ forget to print them and instead use lethals.
> - #4504 

Those are called rubber shot yet they are also bouncy according to the code. This is just rubber shot with no bounce and given to the det roundstart.

**Comment by TeomanTheGreat** [comment]
> Making detectives weapon non-lethal will not discourage detective from performing arrests at all. Also dying doesnt matter in this game you can always be quickly fixed up with no consequences

Are you saying that dets should have lethal in the first place? Because the gun is only for self defense and frankly when det is doing det shit there is no need for him to blast his opponent to smithereens when he can just knock them down with this and cuff em (I would imagine they can arrest a person who just tried to murder them)

**Comment by FriendlyContractor** [comment]
Im saying that your intentions seems to be having detective arrest people less( frankly this whole thing with detective not  supposed to arrest people and being "lesser security" is nonsense but thats for another discussion)  but what you propose is gonna have opposite effect

**Review by BrumalLake — CHANGES_REQUESTED** [review]
identical to bouncy without the ricochet
you should make either a child

**Comment by TeomanTheGreat** [comment]
> identical to bouncy without the ricochet
> you should make either a child

Aight (i dont have much freetime atm but ill get to it when i do)

**Comment by Raven-Industries** [comment]
The opinion of someone who plays det reasonably often: I'm going to be more tempted to make arrests or otherwise 'intervene' if my roundstart ammunition is nonlethals. As it is now, I generally stay hands off because I know that the only real recourse I have is lethals, and since you need solid justification to use them, it means my options are limited. Can't make average arrests, because your only real stopping power also kills them, and that would be overescalation. Can't do searches, because the only way you have to enforce that is your trusty revolver, which you can't use unless they start hacking limbs off. As it stands now, with lethals, I find myself roleplaying more to handle situations because of that restriction (oftentimes I find that I hold to that restriction a little _too_ much and end up dead because of my reluctance to use said revolver).

**Comment by zeskorion** [comment]
> > Seems like a duplicate. We can already make rubber shots at autolathes. Just that detective players ~have no idea how to use an autolathe~ forget to print them and instead use lethals.
> > 
> > * [Adds .38 Rubbers To Autolathes #4504](https://github.com/BeeStation/BeeStation-Hornet/pull/4504)
> 
> Those are called rubber shot yet they are also bouncy according to the code. This is just rubber shot with no bounce and given to the det roundstart.

just give det these and dont add a new one

**Comment by mcmeiler** [comment]
> I'm going to be more tempted to make arrests
> Can't do searches

I won't get into a rules discussion here but as dictated per SOP, detectives aren't allowed to perform arrests and searches *at all* under normal conditions, i.e. green and blue alert and a present security force

**Comment by TeomanTheGreat** [comment]
> 
> 
> identical to bouncy without the ricochet you should make either a child

Done

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Remove the new ammo type and just give them the pre-existing ammo. The path "bouncy" doesn't actually mean they're bouncy. In the original PR they were going to have a ricochet of 6, but it was reset back to 1 before being merged. (They do have a high chance to ricochet though, to be fair.. It's still just once though)

**Comment by Raven-Industries** [comment]
> > I'm going to be more tempted to make arrests
> > Can't do searches
> 
> I won't get into a rules discussion here but as dictated per SOP, detectives aren't allowed to perform arrests and searches _at all_ under normal conditions, i.e. green and blue alert and a present security force

You're right, but my point is that the temptation to just step in _just that once_ and do something is there. I think that temptation would be much greater when you have easy access to a non-lethal option, where it's easier to justify using it because "it won't injure them or anything". Having a purely lethal weapon, in my opinion, forces you to avoid that sort of thing, and RP your way around things rather than disabler-blast a path to your goal. 
To put it another way, I think the detective having a lethal weapon, with strictly limited use cases, rather than a far more loosely restricted non-lethal, facilitates RP by changing the manner in which he/she can interact with people and suspects.

**Comment by 4rchibald** [comment]
Detective is not a better looking security officer, they should not be doing arrests unless it is absolutely necessary. Their job is to investigate, because of that they are often in dangerous places and dealing with dangerous people hence they were issued gun permit for self defence, to be used only in immediate physical threat to them.
This is according to lore and SOP, we just need admins putting more pressure on detectives and heads to enforce it instead of making detective into officer. If there is will to merge this PR then we should simply remove detective and add one more slot in security

**Comment by TeomanTheGreat** [comment]
> Detective is not a better looking security officer, they should not be doing arrests unless it is absolutely necessary. Their job is to investigate, because of that they are often in dangerous places and dealing with dangerous people hence they were issued gun permit for self defence, to be used only in immediate physical threat to them.
> This is according to lore and SOP, we just need admins putting more pressure on detectives and heads to enforce it instead of making detective into officer. If there is will to merge this PR then we should simply remove detective and add one more slot in security 

You can still stop threats to yourself by doing stam damage you know.

**Comment by ghost** [comment]
Detective becomes less of a unique role from security officer every time this kind of change makes it through. If their equipment just becomes reskinned or subpar sec gear then there's no reason to play as a detective when you can just go sec officer and ask for or print a forensic scanner to be a better-geared detective. Giving them nonlethals doesn't add to their job, it's just balancing it around shitters who use lethals when they explicitly aren't supposed to, both IC and as per server rules.

As others have said, not having a nonlethal ranged option (and a subpar melee nonlethal in the form of the wooden baton) makes you play situations differently than you would as a sec officer; you often need people to voluntary let you search or detain them if you try to make an arrest or collect evidence, because the alternative is gunning them down without due cause and getting yourself in trouble with Security or the administration. Having a nonlethal weapon means you can play exactly like a shit sec officer by stunning suspects and rifling through their inventory instead of needing to RP or play smart to do your job.

In addition I don't think it should be possible to print more ammo for the detective's revolver, lethal or nonlethal. Especially speedloaders. Their gun and ammunition should be treated as a valuable (dangerous) and limited resource, not something you can use on a whim and just print more of from the seclathe.

**Comment by TeomanTheGreat** [comment]
Well, that was painful. Now I wonder why this is conflicting.

**Comment by TeomanTheGreat** [comment]
based!

File: code/modules/research/techweb/all_nodes.dm [hunk 1]
```diff
@@ -11,7 +11,7 @@
 	// Default research tech, prevents bricking
 	design_ids = list("basic_matter_bin", "basic_cell", "basic_scanning", "basic_capacitor", "basic_micro_laser", "micro_mani", "desttagger", "handlabel", "packagewrap",
 	"destructive_analyzer", "circuit_imprinter", "experimentor", "rdconsole", "design_disk", "tech_disk", "rdserver", "rdservercontrol", "mechfab", "paystand",
-	"space_heater", "beaker", "large_beaker", "bucket", "xlarge_beaker", "sec_rshot", "sec_beanbag_slug", "sec_bshot", "sec_slug", "sec_Islug", "sec_Brslug", "sec_dart", "sec_38",
+	"space_heater", "beaker", "large_beaker", "bucket", "xlarge_beaker", "sec_rshot", "sec_beanbag_slug", "sec_bshot", "sec_slug", "sec_Islug", "sec_Brslug", "sec_dart", "sec_38" "sec_38b",
```
Comment (Archanial) [inline code/modules/research/techweb/all_nodes.dm]:
```suggestion
	"space_heater", "beaker", "large_beaker", "bucket", "xlarge_beaker", "sec_rshot", "sec_beanbag_slug", "sec_bshot", "sec_slug", "sec_Islug", "sec_Brslug", "sec_dart", "sec_38", "sec_38b",
```

**Review by PowerfulBacon — DISMISSED** [review]
Use a regular rubber type rather than the bouncy bullets

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
Apparently the bouncy bullets are bouncy, I'd rather you used non-bouncy bullets.
